bitcoin/src/consensus/mod.rs

```diff
@@ -16,7 +16,7 @@ pub mod validation;
 #[rustfmt::skip]                // Keep public re-exports separate.
 #[doc(inline)]
 pub use self::{
-    encode::{deserialize, deserialize_partial, serialize, Decodable, Encodable, ReadExt, WriteExt},
+    encode::{deserialize, deserialize_partial, serialize, serialize_hex, Decodable, Encodable, ReadExt, WriteExt, CheckedData, MAX_VEC_SIZE, Error},
```

Comment:
`ChckedData` is very rare, isn't it?

ACK `serialize_hex`.

`consensus::encode::Error` should be renamed to `DecodeError` especially since there's an error type in validation.

But I'm thinking, since validation only validates transactions we should move it to `transaction`.

Reply:
We have a refactor of the `consensus` module in flight (#2212) and in that one I made the error `consensus::Error` but its waiting on you before we can push forwards with it.

Reply:
Yeah, I think renaming `consensus` to `consensus_encoding` and move `validation` to `transaction` is a good approach.

Reply:
https://github.com/rust-bitcoin/rust-bitcoin/pull/2406